acsoo/addons.py

```diff
@@ -85,3 +92,93 @@ def addons_list_depends(ctx, exclude):
 
 
 addons.add_command(addons_list_depends, 'list-depends')
+
+
+@click.command(help="Print a comma separated list of the modified installable "
+                    "addons from the given git ref")
+@click.argument('git_ref')
+@click.option('-r', '--diff-requirements', 'diff_requirements', is_flag=True,
+              help="Defines whether the comparison must take the requirements "
+                   "file into account or not.")
+@click.pass_context
+def addons_toupdate(ctx, git_ref, diff_requirements):
+    # Check ancestor
+    if call(['git', 'merge-base', '--is-ancestor', 'HEAD', git_ref]):
+        click.echo('all')
+        return
+    addon_names = []
+    # Compare each installable addons and populate modified addons list
+    addons_paths = ctx.obj['addons_paths']
+    for addon_name in addons_paths:
+        addon_dir = os.path.join(addons_paths[addon_name], addon_name)
+        if call(['git', 'diff', '--quiet', git_ref, addon_dir]):
+            addon_names.append(addon_name)
+    # Requirements file comparison
+    if diff_requirements:
+        req_ref = 'HEAD:requirements.txt'
```

Comment:
@sbidoul Using HEAD won't manage uncommitted changes. Should we read the current requirements file or use its last version in the last commit ? And if we use HEAD, do we check if there are uncommitted changes ?

Reply:
It should be the same comparison as for the addons. 

For the addons, you do `git diff` so I guess reading the local current requirements.txt will give the same result?